### ISSUE TITLE
Am I Using Masked Images Feature Correctly?

### ISSUE THREAD

**Issue by Dazrock** [derrian-distro/LoRA_Easy_Training_Scripts#331]
Hey guys. 
I'm worried that i might be using the Masked Loss function incorrectly. Because i tried to check to see if there is any sign of it working in the LoRa Trainer's console, and i can't see anything related to it popping up before it starts training. 

I have the base directory with all my images, and captions. (This works fine)
I tried to find sources on masked training using (LoRA Easy Training Scripts), and couldn't find anything. 
So i tried experimenting by making masks manually, and matching the name of the masked image to each of their relative training image names.
I then had my main folder with the training images + captions, and a separate folder with just the mask images in them.

I enabled Mask Loss, and it appears to train with no errors at all. The result of the finished LoRa acts as if it wasn't using the masks at all. 
So i was wondering if you guys use it/set it up this way too? 

If there is a source for how to use the feature within LoRA Easy Training Scripts i'd very much appreciate it if someone knows of one.

Thank you for your time.

**Comment by Jelosus2** [comment]
Yes, each mask has to match the file name of the respective image. And when you enable Masked Loss be sure to load the directory where you have the masks along side with the one you have the dataset in the subset

**Comment by Dazrock** [comment]
Thank you so much for the reply. 

Yeah, i have (Subset 1) with the main directory of training images with the caption files, and a completely separate directory path for the masked images pasted in the Masked Directory path box. 

If this is how you use them, that is great to know. I just wasn't sure if i was doing it right, because i can't see any indicator of it detecting masked images in the console. 
Then with my end results being worse than expected i wasn't sure if it was working or not.

 Thanks again for replying, I guess it's working as intended then.

**Comment by Dazrock** [comment]
> Yes, each mask has to match the file name of the respective image. And when you enable Masked Loss be sure to load the directory where you have the masks along side with the one you have the dataset in the subset

Thank you so much for the reply.

Yeah, i have (Subset 1) with the main directory of training images with the caption files, and a completely separate directory path for the masked images pasted in the Masked Directory path box.

If this is how you use them, that is great to know. I just wasn't sure if i was doing it right, because i can't see any indicator of it detecting masked images in the console.
Then with my end results being worse than expected i wasn't sure if it was working or not.

One more question, if you don't mind. I assume the correct colors to use for the masks are Black, and white? 
(Black) to ignore, and (White) to focus? 

Thanks again for replying, I guess it's working as intended then?

**Comment by Jelosus2** [comment]
I think they only need to be 2 different colors. I personally use black to ignore and red to mask

**Comment by Jelosus2** [comment]
Closing this issue due to inactivity, feel free to open a new issue in the future
